azure-iot-device/azure/iot/device/common/pipeline/pipeline_stages_mqtt.py

```diff
@@ -176,3 +210,32 @@ def _on_message_received(self, topic, payload):
             stage=self,
             event=pipeline_events_mqtt.IncomingMQTTMessageEvent(topic=topic, payload=payload),
         )
+
+    def _on_unexpected_connection(self):
+        """
+        Handler that gets called by the transport when it connections.  This handler only
+        gets called when it connects outside of the normal connect/reconnect logic.
+        """
+        logger.warn("an unexpected connection occured")
+        self.on_connected()
+
+    def _on_unexpected_connection_failure(self, cause):
```

Comment:
>_on_unexpected_connection_failure [](start = 8, length = 33)

This one is a little stranger.  In this case, Paho is saying, "that connection that you asked for failed", when we didn't ask for a connection (or we gave up).  It seems like passing the error up to the unhandled_error_handler is the best case.  Improvement suggestion: wrap this in an "unexpected connection failure" error, so the user can see an object that specifically says "there was an unexpected connection failure caused by an auth failure".  This seems more useful than just passing the auth failure up without the wrapper that indicates that it just might be useless because things are a little wonky anyway.

Reply:
I agree: put that in a `TODO:`

---
In reply to: [302272141](https://github.com/Azure/azure-iot-sdk-python-preview/pull/136#discussion_r302272141) [](ancestors = 302272141)